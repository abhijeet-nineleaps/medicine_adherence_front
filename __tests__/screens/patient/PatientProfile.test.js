import React from 'react';
import renderer from 'react-test-renderer';
import enableHooks from "jest-react-hooks-shallow";
import ViewProfile from '../../../src/screens/patient/PatientProfile';
import Enzyme from 'enzyme';
import Adapter from '@wojtekmaj/enzyme-adapter-react-17';
Enzyme.configure({adapter: new Adapter()});
enableHooks(jest);
jest.mock("react-redux", () => ({
  ...jest.requireActual("react-redux"),
  useSelector: jest.fn(),
  useDispatch: jest.fn(),
}));
jest.mock("@react-navigation/native", () => ({
  ...jest.requireActual("@react-navigation/native"),
  useFocusEffect: jest.fn().mockImplementation((func) => func()),
}));
describe('Click send image', () => {
  it('renders correctly', () => {
    const tree = renderer
      .create(<ViewProfile route={undefined} navigation={undefined}/>)
      .toJSON();
    expect(tree).toMatchSnapshot();
  });
  
});