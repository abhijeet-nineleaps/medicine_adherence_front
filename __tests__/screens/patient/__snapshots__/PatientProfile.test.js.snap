// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Click send image renders correctly 1`] = `
<View
  style={
    Object {
      "backgroundColor": "white",
      "height": "100%",
    }
  }
>
  <View
    style={
      Object {
        "backgroundColor": "white",
        "flexDirection": "column",
        "height": "100%",
      }
    }
  >
    <RCTScrollView>
      <View>
        <View
          style={
            Object {
              "flexDirection": "row",
              "height": 110,
              "justifyContent": "space-between",
              "padding": 10,
            }
          }
        >
          <View
            style={
              Object {
                "flexDirection": "column",
              }
            }
          >
            <Text
              style={
                Object {
                  "paddingLeft": 5,
                  "paddingTop": 9,
                }
              }
            >
              Name
            </Text>
            <Text
              style={
                Object {
                  "color": "black",
                  "fontSize": 17,
                  "padding": 5,
                }
              }
            />
          </View>
          <View
            style={
              Object {
                "justifyContent": "flex-end",
              }
            }
          >
            <Image
              source={
                Object {
                  "uri": undefined,
                }
              }
              style={
                Object {
                  "borderColor": "white",
                  "borderRadius": 45,
                  "borderWidth": 2,
                  "height": 90,
                  "width": 90,
                }
              }
            />
          </View>
        </View>
        <View
          style={
            Object {
              "borderBottomColor": "grey",
              "borderBottomWidth": 0.3,
            }
          }
        />
        <View
          style={
            Object {
              "flexDirection": "row",
              "justifyContent": "space-between",
              "marginBottom": 12,
              "padding": 15,
            }
          }
        >
          <Text
            style={Object {}}
          >
            Bio
          </Text>
          <Text
            style={
              Object {
                "color": "black",
                "width": 200,
              }
            }
          />
        </View>
        <View
          style={
            Object {
              "borderBottomColor": "grey",
              "borderBottomWidth": 0.3,
            }
          }
        />
        <View
          style={
            Object {
              "flexDirection": "row",
              "justifyContent": "space-between",
              "marginBottom": 12,
              "padding": 15,
            }
          }
        >
          <Text
            style={Object {}}
          >
            Contact Number
          </Text>
          <Text
            style={
              Object {
                "color": "black",
                "width": 200,
              }
            }
          />
        </View>
        <View
          style={
            Object {
              "borderBottomColor": "grey",
              "borderBottomWidth": 0.3,
            }
          }
        />
        <View
          style={
            Object {
              "flexDirection": "row",
              "justifyContent": "space-between",
              "marginBottom": 12,
              "padding": 15,
            }
          }
        >
          <Text
            style={Object {}}
          >
            Email Id
          </Text>
          <Text
            style={
              Object {
                "color": "black",
                "width": 200,
              }
            }
          />
        </View>
        <View
          style={
            Object {
              "borderBottomColor": "grey",
              "borderBottomWidth": 0.3,
            }
          }
        />
        <View
          style={
            Object {
              "flexDirection": "row",
              "justifyContent": "space-between",
              "marginBottom": 12,
              "padding": 15,
            }
          }
        >
          <Text
            style={Object {}}
          >
            Gender
          </Text>
          <Text
            style={
              Object {
                "color": "black",
                "width": 200,
              }
            }
          />
        </View>
        <View
          style={
            Object {
              "borderBottomColor": "grey",
              "borderBottomWidth": 0.3,
            }
          }
        />
        <View
          style={
            Object {
              "flexDirection": "row",
              "justifyContent": "space-between",
              "marginBottom": 12,
              "padding": 15,
            }
          }
        >
          <Text
            style={Object {}}
          >
            Blood Group
          </Text>
          <Text
            style={
              Object {
                "color": "black",
                "width": 200,
              }
            }
          />
        </View>
        <View
          style={
            Object {
              "borderBottomColor": "grey",
              "borderBottomWidth": 0.3,
            }
          }
        />
        <View
          style={
            Object {
              "flexDirection": "row",
              "justifyContent": "space-between",
              "marginBottom": 12,
              "padding": 15,
            }
          }
        >
          <Text
            style={Object {}}
          >
            Marital Status
          </Text>
          <Text
            style={
              Object {
                "color": "black",
                "width": 200,
              }
            }
          />
        </View>
        <View
          style={
            Object {
              "borderBottomColor": "grey",
              "borderBottomWidth": 0.3,
            }
          }
        />
        <View
          style={
            Object {
              "flexDirection": "row",
              "justifyContent": "space-between",
              "marginBottom": 12,
              "padding": 15,
            }
          }
        >
          <Text
            style={Object {}}
          >
            Age(in yrs)
          </Text>
          <Text
            style={
              Object {
                "color": "black",
                "width": 200,
              }
            }
          />
        </View>
        <View>
          <View
            style={
              Array [
                Object {
                  "marginVertical": 8,
                },
                Object {
                  "backgroundColor": "white",
                },
              ]
            }
            theme={
              Object {
                "animation": Object {
                  "scale": 1,
                },
                "colors": Object {
                  "accent": "#03dac4",
                  "backdrop": "rgba(0, 0, 0, 0.5)",
                  "background": "#f6f6f6",
                  "disabled": "rgba(0, 0, 0, 0.26)",
                  "error": "#B00020",
                  "notification": "#f50057",
                  "onSurface": "#000000",
                  "placeholder": "rgba(0, 0, 0, 0.54)",
                  "primary": "#6200ee",
                  "surface": "#ffffff",
                  "text": "#000000",
                },
                "dark": false,
                "fonts": Object {
                  "light": Object {
                    "fontFamily": "System",
                    "fontWeight": "300",
                  },
                  "medium": Object {
                    "fontFamily": "System",
                    "fontWeight": "500",
                  },
                  "regular": Object {
                    "fontFamily": "System",
                    "fontWeight": "400",
                  },
                  "thin": Object {
                    "fontFamily": "System",
                    "fontWeight": "100",
                  },
                },
                "roundness": 4,
              }
            }
          >
            <View>
              <View
                style={
                  Object {
                    "backgroundColor": "#f6f6f6",
                  }
                }
              >
                <View
                  accessibilityRole="button"
                  accessibilityState={
                    Object {
                      "disabled": false,
                      "expanded": false,
                    }
                  }
                  accessible={true}
                  focusable={true}
                  onClick={[Function]}
                  onResponderGrant={[Function]}
                  onResponderMove={[Function]}
                  onResponderRelease={[Function]}
                  onResponderTerminate={[Function]}
                  onResponderTerminationRequest={[Function]}
                  onStartShouldSetResponder={[Function]}
                  style={
                    Array [
                      Object {
                        "overflow": "hidden",
                      },
                      Array [
                        Object {
                          "padding": 8,
                        },
                        undefined,
                      ],
                    ]
                  }
                >
                  <View
                    pointerEvents="none"
                    style={
                      Object {
                        "alignItems": "center",
                        "flexDirection": "row",
                      }
                    }
                  >
                    <Text
                      allowFontScaling={false}
                      selectable={false}
                      style={
                        Array [
                          Object {
                            "color": "black",
                            "fontSize": 16,
                          },
                          Object {
                            "marginLeft": 8,
                          },
                          Object {
                            "fontFamily": "FontAwesome",
                            "fontStyle": "normal",
                            "fontWeight": "normal",
                          },
                          Object {},
                        ]
                      }
                    >
                      
                    </Text>
                    <View
                      style={
                        Array [
                          Object {
                            "margin": 8,
                          },
                          Object {
                            "flex": 1,
                            "justifyContent": "center",
                          },
                        ]
                      }
                    >
                      <Text
                        numberOfLines={1}
                        selectable={false}
                        style={
                          Array [
                            Object {
                              "color": "#000000",
                              "fontFamily": "System",
                              "fontWeight": "400",
                            },
                            Object {
                              "textAlign": "left",
                            },
                            Array [
                              Object {
                                "fontSize": 16,
                              },
                              Object {
                                "color": "rgba(0, 0, 0, 0.87)",
                              },
                              Object {
                                "fontSize": 15,
                                "fontWeight": "500",
                              },
                            ],
                          ]
                        }
                      >
                        Medicines
                      </Text>
                    </View>
                    <View
                      style={
                        Array [
                          Object {
                            "margin": 8,
                          },
                          undefined,
                        ]
                      }
                    >
                      <Text
                        allowFontScaling={false}
                        selectable={false}
                        style={
                          Array [
                            Object {
                              "color": undefined,
                              "fontSize": 12,
                            },
                            undefined,
                            Object {
                              "fontFamily": "FontAwesome",
                              "fontStyle": "normal",
                              "fontWeight": "normal",
                            },
                            Object {},
                          ]
                        }
                      >
                        
                      </Text>
                    </View>
                  </View>
                </View>
              </View>
            </View>
          </View>
        </View>
      </View>
    </RCTScrollView>
  </View>
</View>
`;
